### PR TITLE
Enhanced care config to validate env during build process

### LINKED ISSUE #8470: Enhancements to care config
https://github.com/ohcnetwork/care_fe/pull/8445 Introduces `care.config.ts` for making CARE configurable from ENVs. However there are couple of enhancements that can be done:

- [ ] Perform config validation during vite build so that build fails thus preventing exceptions from being thrown during runtime if any.
- [ ] Sentry config to be optional and enable sentry only if configured.
- [ ] Plausible config to be optional and enable plausible only if configured.

### PR DESCRIPTION
**Pull request by ghost**
## Proposed Changes

- Fixes #8470 
- Vite build process now includes validation of the required environment variables using the @julr/vite-plugin-validate-env plugin
- Sentry & Plausible are optional & will only be enabled if env are configured correctly

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Added new dependencies for environment validation (`@julr/vite-plugin-validate-env` and `zod`).
	- Integrated environment validation within the configuration setup to ensure required variables are defined.

- **Bug Fixes**
	- Enhanced error handling in the `Plausible` and `Sentry` components to ensure correct configuration before execution.

### CODE REVIEW

**Comment by Alokih** [comment]
@Jacobjeevan sir, I have created a PR can you review it once , I have just added validation for only 5 env , should I add more ?

File: src/Integrations/Plausible.tsx [hunk 1]
```diff
@@ -10,6 +10,13 @@ export default function Plausible() {
   });
 
   useEffect(() => {
+    if (!careConfig.plausible.domain || !careConfig.plausible.server) {
+      console.error(
+        "Plausible is not configured correctly. Please check your environment variables.",
+      );
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/Integrations/Plausible.tsx]: Go ahead and make the changes suggested by coderabbit @Alokih

**Comment by nihal467** [comment]
@Alokih can you fix the lint failure

**Comment by Alokih** [comment]
@nihal467 sir I just updated the branch is it ok now ?

**Comment by ghost** [comment]
@nihal467 sir , I updated my node version & ran npm run install-all again to update lock file. 
Is this correct ? 
If not , can you please help me fixing lint errors ?

**Comment by nihal467** [comment]
build, cypress and random check looks good, pushing it for code review

**Review by khavinshankar — APPROVED** [review]

**Comment by khavinshankar** [comment]
@rithviknishad @Jacobjeevan can u please review this PR

**Review by Jacobjeevan — APPROVED** [review]

**Pull request merged**